tools/torch/data.lua

```diff
@@ -41,33 +41,205 @@ local function all_keys(cursor_,key_,op_)
         end)
 end
 
-local PreProcess = function(y, meanTensor, mirror, crop, train, cropY, cropX, croplen)
+-- HSV Augmentation
+-- Parameters:
+-- @param im (tensor): input image
+-- @param augHSV (table): standard deviations under {H,S,V} keys with float values.
+local augmentHSV = function(im_rgb, augHSV)
+    -- Fair augHSV standard deviation values are {H=0.02,S=0.04,V=0.08}
+    local im_hsv = image.rgb2hsv(im_rgb)
+    if augHSV.H >0.001 then
+        -- We do not need to account for overflow because every number wraps around (1.1=2.1=3.1,etc)
+        local rng_h = 1 + torch.normal(0, augHSV.H) -- We add a round value 1 to prevent an underflow bug (<0 becomes black bug)
+        im_hsv[1] = torch.add(im_hsv[1], torch.FloatTensor(1,im_hsv:size(2),im_hsv:size(2)):fill(rng_h))
```

Comment:
Good! I think that's more efficient. I'll have to add :float() to the scalar though; the color transformation cries about double's.